File: java_SDK/PayHubSDK/src/main/java/com/payhub/ws/api/CaptureResponseInformation.java
package com.payhub.ws.api;
import java.io.IOException;
import java.util.List;

import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.core.JsonParseException;
import com.fasterxml.jackson.databind.JsonMappingException;
import com.payhub.ws.model.CaptureResponse;



@JsonInclude(JsonInclude.Include.NON_NULL)  
public class CaptureResponseInformation {
	
    private CaptureResponse lastCaptureResponse;
    public Object _links;
    private List<Errors> errors;
    public String rowData;
    private Object metaData;
    private TransactionManager transactionManager;
    private BillInformation billInformation;
    private MerchantInformation merchantInformation; 
    
	public String getMetaData() {
		return (String) metaData;
	}
	public void setMetaData(Object metadata) {
		this.metaData = metadata;
	}
	public CaptureResponse getLastCaptureResponse() {
		return lastCaptureResponse;
	}
	public void setLastCaptureResponse(CaptureResponse lastCaptureResponse) {
		this.lastCaptureResponse = lastCaptureResponse;
	}
	public Object get_links() {
		return _links;
	}
	public void set_links(Object _links) {
		this._links = _links;
	}
	public List<Errors> getErrors() {
		return errors;
	}
	public void setErrors(List<Errors> errors) {
		this.errors = errors;
	}
	public String getRowData() {
		return rowData;
	}
	public void setRowData(String rowData) {
		this.rowData = rowData;
	}
	/**
	 * @param transactionManager the transactionManager to set
	 */
	public void setTransactionManager(TransactionManager transactionManager) {
		this.transactionManager = transactionManager;
	}
	/**
	 * @return the billInformation
	 * @throws IOException 
	 * @throws JsonMappingException 
	 * @throws JsonParseException 
	 */
	public BillInformation getBillInformation() throws JsonParseException, JsonMappingException, IOException {
		if(billInformation==null){
			BillInformation b = new BillInformation(this.transactionManager);
			b.setUrl(this.transactionManager.getUrl()+"capture/");
			b.getBillForRecurringBillInformationById(lastCaptureResponse.getTransactionId());
			billInformation=b;
		}
		return billInformation;
	}
	/**
	 * @return the merchantInformation
	 * @throws IOException 
	 * @throws JsonMappingException 
	 * @throws JsonParseException 
	 */
	public MerchantInformation getMerchantInformation() throws JsonParseException, JsonMappingException, IOException {
		if(merchantInformation==null){			
				MerchantInformation m = new MerchantInformation(this.transactionManager);
				m.getDataByTransaction(TransactionType.Capture, lastCaptureResponse.getTransactionId());
				merchantInformation=m;				
			}
		return merchantInformation;
	}
}

File: Payhub - SDK/php_SDK/com/payhub/ws/api/CaptureResponseInformation.php
<?php

/**
 * Created by PhpStorm.
 * User: agustin
 * Date: 24/07/2015
 * Time: 10:08
 */
class CaptureResponseInformation
{
    public $lastCaptureResponse;
    public $_links;
    public $errors;
    public $rowData;
    public $metaData;
    private $transactionManager;
    private $billInformation;
    private $merchantInformation;

    /**
     * @return mixed
     */
    public function getLastCaptureResponse()
    {
        return $this->lastCaptureResponse;
    }

    /**
     * @param mixed $lastCaptureResponse
     */
    public function setLastCaptureResponse($lastCaptureResponse)
    {
        $this->lastCaptureResponse = $lastCaptureResponse;
    }

    /**
     * @return mixed
     */
    public function getLinks()
    {
        return $this->_links;
    }

    /**
     * @param mixed $links
     */
    public function setLinks($links)
    {
        $this->_links = $links;
    }

    /**
     * @return mixed
     */
    public function getErrors()
    {
        return $this->errors;
    }

    /**
     * @param mixed $errors
     */
    public function setErrors($errors)
    {
        $this->errors = $errors;
    }

    /**
     * @return mixed
     */
    public function getRowData()
    {
        return $this->rowData;
    }

    /**
     * @param mixed $rowData
     */
    public function setRowData($rowData)
    {
        $this->rowData = $rowData;
    }

    /**
     * @return mixed
     */
    public function getMetaData()
    {
        return $this->metaData;
    }

    /**
     * @param mixed $metaData
     */
    public function setMetaData($metaData)
    {
        $this->metaData = $metaData;
    }

    /**
     * @param mixed $transactionManager
     */
    public function setTransactionManager($transactionManager)
    {
        $this->transactionManager = $transactionManager;
    }

    /**
     * @return mixed
     */
    public function getBillInformation()
    {
        if($this->billInformation==null){
            $b = new BillInformation($this->transactionManager);
			$b->setUrl($this->transactionManager->getUrl()."capture/");
			$b->getBillForSaleInformationByTransactionId($this->lastCaptureResponse->getTransactionId());
			$this->billInformation=$b;
		}
        return $this->billInformation;
    }

    /**
     * @return mixed
     */
    public function getMerchantInformation()
    {
        if($this->merchantInformation==null){
            $m = new MerchantInformation($this->transactionManager);
				$m->getDataByTransaction(TransactionType::Capture, $this->lastCaptureResponse->getTransactionId());
				$this->merchantInformation=$m;
			}
        return $this->merchantInformation;
    }

    public static function fromArray($data){
        $capture = new CaptureResponseInformation();

        foreach ($data as $key => $value){
            if( property_exists( get_class($capture), $key ) ) {
                if($key=="lastCaptureResponse"){
                    $capture->{$key}=CaptureResponse::fromArray($value);
                }elseif($key=="errors"){
                    $capture->{$key}=Errors::fromArray($value);
                }
                else{
                    $capture->{$key} = $value;
                }
            }
        }
        return $capture;
    }
}